### PR TITLE
#253 Polls

### PR DESCRIPTION
**Pull request by Goff-Davis**
# Proposed Changes

Add Polls API coverage.

- Add polls.py
- Add polls functions to canvas.py
- Add polls unit tests and endpoint

Fixes issue #253

### CODE REVIEW

File: canvasapi/poll.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from canvasapi.util import combine_kwargs
+
+
+@python_2_unicode_compatible
+class Poll(CanvasObject):
+
+    def __str__(self):
+        return "{} ({})".format(self.question, self.id)
+
+    def update(self, question, description, **kwargs):
```
Comment (Thetwam) [inline canvasapi/poll.py]: Same as above, should be `poll`, not `question` and `description`

File: tests/fixtures/poll.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
+	"create_poll": {
+		"method": "POST",
+		"endpoint": "polls",
+		"data": {
```
Comment (Thetwam) [inline tests/fixtures/poll.json]: According to the docs, this returns a response that looks something like this

```
{
  "polls": [Poll]
}
```

Confirm this with a manual call to 2c.test and/or dev Canvas and adapt the fixture accordingly.

File: tests/fixtures/poll.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+				"created_at": "2014-01-07T15:18:12Z",
+				"user_id": 1
+			}
+		],
+		"status_code": 200
+	},
+	"update": {
+		"method": "PUT",
+		"endpoint": "polls/1",
+		"data": {
```
Comment (Thetwam) [inline tests/fixtures/poll.json]: Same as above, this should return something that looks like

```
{
  "polls": [Poll]
}
```

File: tests/fixtures/poll.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			"question": "Is this a question?",
+			"description": "This is a test.",
+			"created_at": "2014-01-07T13:10:19Z"
+		},
+		"status_code": 200
+	},
+	"get_polls": {
+		"method": "GET",
+		"endpoint": "polls",
+		"data": [
```
Comment (Thetwam) [inline tests/fixtures/poll.json]: Check response on this, may have a `"polls"` keyword

File: tests/fixtures/poll.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	},
+	"delete_poll": {
+		"method": "DELETE",
+		"endpoint": "polls/1",
+		"status_code": 204
+	},
+	"get_poll": {
+		"method": "GET",
+		"endpoint": "polls/1",
+		"data": {
```
Comment (Thetwam) [inline tests/fixtures/poll.json]: Check response.

**Comment by Goff-Davis** [comment]
The issues should be fixed now.

**Comment by Goff-Davis** [comment]
Looks like I forgot to run flake8. That's embarrassing.

**Comment by Goff-Davis** [comment]
This should be ok now. Sorry for all the extraneous commits.

**Pull request merged**